src/background/dnr.js

```diff
@@ -42,6 +49,15 @@ if (__PLATFORM__ !== 'firefox') {
       );
     }
 
+    // Add redirect protection if enabled and any blocking is active
+    if (
+      ids.length &&
+      options.redirectProtection?.enabled &&
```

Comment:
Pointless `?.` - options model has structure, that allways requires fields to be present.